test/e2e/performanceprofile/functests/13_llc/llc.go

```diff
@@ -486,73 +490,366 @@ var _ = Describe("[rfe_id:77446] LLC-aware cpu pinning", Label(string(label.Open
+					case partialAlignment:
+						// WIth 2 guaranteed containers, where 1 is requesting half of L3CacheGroupSize
+						// and another container requesting 2 less than the remaining cpus
+						cpuSize = L3CacheGroupSize / 2
+						cpuResources := []string{fmt.Sprintf("%d", cpuSize-2)}
+						containerList = createMultipleContainers(containerName, cpuResources)
+					}
+					rl := &corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse(strconv.Itoa(cpuSize)),
+						corev1.ResourceMemory: resource.MustParse("100Mi"),
+					}
+
+					p := makePod(testutils.NamespaceTesting, withRequests(rl), withLimits(rl))
+					nodeSelector["kubernetes.io/hostname"] = targetNode.Name
+					dp := deployments.Make(deploymentName, testutils.NamespaceTesting,
+						deployments.WithPodTemplate(p),
+						deployments.WithNodeSelector(nodeSelector))
+					dp.Spec.Selector.MatchLabels = podLabel
+					dp.Spec.Template.ObjectMeta.Labels = podLabel
+					dp.Spec.Template.Spec.Containers = append(dp.Spec.Template.Spec.Containers, containerList...)
+					err = testclient.Client.Create(ctx, dp)
+					Expect(err).ToNot(HaveOccurred())
+					defer func() {
+						testlog.TaggedInfof("Cleanup", "Deleting Deployment %v", deploymentName)
+						err := testclient.DataPlaneClient.Delete(ctx, dp)
+						Expect(err).ToNot(HaveOccurred())
+						waitForDeploymentPodsDeletion(ctx, &targetNode, podLabel)
+					}()
+					podList := &corev1.PodList{}
+					listOptions := &client.ListOptions{Namespace: testutils.NamespaceTesting, LabelSelector: labels.SelectorFromSet(dp.Spec.Selector.MatchLabels)}
+					Eventually(func() bool {
+						isReady, err := deployments.IsReady(ctx, testclient.Client, listOptions, podList, dp)
+						Expect(err).ToNot(HaveOccurred())
+						return isReady
+					}, time.Minute, time.Second).Should(BeTrue())
+					Expect(testclient.Client.List(ctx, podList, listOptions)).To(Succeed())
+					Expect(len(podList.Items)).To(Equal(1), "Expected exactly one pod in the list")
+					testpod := podList.Items[0]
+					err = getter.Container(ctx, &testpod, "test", cpusetCfg)
+					Expect(err).ToNot(HaveOccurred())
+					containerWithIntegralCpu, err := cpuset.Parse(cpusetCfg.Cpus)
+					Expect(err).ToNot(HaveOccurred())
+					// fetch ccx to which cpu used by pod is part of
+					L3CacheGroupCpus, err := getCCX(containerWithIntegralCpu.List()[0])
+					Expect(err).ToNot(HaveOccurred())
+					Expect(containerWithIntegralCpu.IsSubsetOf(L3CacheGroupCpus)).To(BeTrue())
+					if alignmentType == partialAlignmentWithContainers {
+						expectedBurstablePodCpus := totalOnlineCpus.Difference(containerWithIntegralCpu)
+						err = getter.Container(ctx, &testpod, "log1", cpusetCfg)
+						Expect(err).ToNot(HaveOccurred())
+						containerWithnonIntegralCpu1, err := cpuset.Parse(cpusetCfg.Cpus)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(containerWithnonIntegralCpu1.Equals(expectedBurstablePodCpus)).To(BeTrue())
```

Comment:
for these types of assertions I always prefer to print out the values of each variable if the assert failed to enhance the troubleshooting, here and in similar assertions in this PR. the reason for this is that when such Expect() fails one can only know that "Expected false to equal true" and no other details is provided about the actual compared values.